ros2topic/ros2topic/verb/echo.py

```diff
@@ -99,14 +154,10 @@ def main(self, *, args):
         self.no_arr = args.no_arr
         self.no_str = args.no_str
 
-        qos_profile = qos_profile_from_short_keys(
-            args.qos_profile,
-            reliability=args.qos_reliability,
-            durability=args.qos_durability,
-            depth=args.qos_depth,
-            history=args.qos_history)
-
         with NodeStrategy(args) as node:
+
+            qos_profile = self.chooseQoS(node, args)
```

Comment:
ideally, if the user provided an argument that should be preferred over the autodetection logic.
I mean one of these:

https://github.com/ros2/ros2cli/blob/56db2ba9fcc85ead292b0c3403cf27dba41854e2/ros2topic/ros2topic/api/__init__.py#L189-L219

@jacobperron what do you think?

Reply:
to handle that I would delete the defaults from the argument and make them optional, so you can check if they were set or not

Reply:
I agree. The CLI options should have priority over the auto-detection logic.

Reply:
I made some changes to give CLI options priority over auto-detection logic

Addresed in [8efa6ed](https://github.com/ros2/ros2cli/pull/613/commits/8efa6ede28b8f5a77c912a5ae690f0cf5db7ba75)